libraries/WiFi/src/WiFiMulti.cpp

```diff
@@ -261,12 +269,10 @@ bool WiFiMulti::testConnection(int32_t i){
     WiFiClient client;
     const int httpPort = 80;
     if (!client.connect(host.c_str(), httpPort)) {
-        log_e("connection failed");
-        Serial.println("connection failed");
-        markAsFailed(i);
+        log_e("Connection failed");
         return false;
     } else {
-        log_i("connection to test host successful");
+        log_i("Connected to test host");
```

Comment:
at this point it should be possible to close the connection without reading the actual data

Reply:
See above.  I would rather skip the text check to make it simpler.  Just need to make sure we don't get trapped.  If this device gets onto xfinitywifi here it doesn't want to get off.

Reply:
yeah, ideally this would exit once it receives the initial http header details (all lines until a blank line IIRC). Once the header has been received the connection can be closed. I'd recommend read by line and check the line for known keywords (Location: XXX, HTTP XXX 200|302 etc...) That way you can abort earlier from the line by line checking.

Reply:
I agree with @atanisoft here :) there are some codes that show all is good.

Reply:
actually what is the target? which codes would mean that something went bad?

Reply:
All I am trying to do is ensure that the device can actually reach the internet - captive portals are the problem.  If you're connected to Wifi but behind a captive portal you're not going anywhere.  This functionality is most useful when you have bAllowOpenAP, which automatically connects to any and all open APs in addition to the ones in your list just to get online.  In my case, once it's online it connects to my server, grabs a config file with the correct SSID information then turns bAllowOpenAP off and bStrictMode on.  It's good functionality to put 1000 brand new devices online just by plugging them in and setting your AP to open mode for 15 minutes.

I'm looking for anything but a 304 by default, but if that isn't good enough, you can set your own URL and look for any test phrase that you want, including text in a webpage.  By default I don't want to look for 200's either in case a captive portal happens to return that.  Looking for a permanent redirect seems like the best solution because a captive portal wouldn't do that and a site like google or amazon should be available everywhere.